pkg/datasource/csvsource/csv_kline_reader_test.go

```diff
@@ -0,0 +1,161 @@
+package csvsource
+
+import (
+	"encoding/csv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+func assertKLineEq(t *testing.T, exp, act types.KLine, name string) {
+	assert.Equal(t, exp.StartTime, act.StartTime, name)
+	assert.Equal(t, 0, exp.Open.Compare(act.Open), name)
+	assert.Equal(t, 0, exp.High.Compare(act.High), name)
+	assert.Equal(t, 0, exp.Low.Compare(act.Low), name)
+	assert.Equal(t, 0, exp.Close.Compare(act.Close), name)
+	assert.Equal(t, 0, exp.Volume.Compare(act.Volume), name)
+}
+
+func TestCSVKLineReader_ReadWithBinanceDecoder(t *testing.T) {
+	tests := []struct {
+		name string
+		give string
+		want types.KLine
+		err  error
+	}{
+		{
+			name: "Read DOHLCV",
+			give: "1609459200000,28923.63000000,29031.34000000,28690.17000000,28995.13000000,2311.81144500",
+			want: types.KLine{
+				StartTime: types.NewTimeFromUnix(1609459200, 0),
+				Open:      fixedpoint.NewFromFloat(28923.63),
```

Comment:
you can now dot import the testhelper for these fixed points

check pkg/testing/testhelper/number.go 

and check pkg/strategy/liquiditymaker/generator_test.go for example

Reply:
noted thx